lib/linter/linter.js

```diff
@@ -922,6 +922,9 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserOptions, parser
                         if (problem.fix && rule.meta && !rule.meta.fixable) {
                             throw new Error("Fixable rules should export a `meta.fixable` property.");
                         }
+                        if (problem.suggestions && !(rule.meta && rule.meta.docs && rule.meta.docs.suggestion === true)) {
+                            throw new Error("Rules with suggestions should set the `meta.docs.suggestion` property to `true`.");
+                        }
```

Comment:
There are not many details or discussions about this property in the [RFC](https://github.com/eslint/rfcs/pull/30), but it was probably added to `docs` instead of as a top-level property because it was never meant to be mandatory, even not for rules that provide suggestions. I believe the purpose was only to flag those rules on our website. The whole `docs` object is optional for third-party rules, so it doesn't look to me that this should throw an error.

`fixable` is different, its value was supposed to be used for some optimizations.

Reply:
I see. Yeah it doesn't seem like there was much discussion about it. I personally would have chosen one of the following different designs:

1. Add a new boolean property `meta.suggestable` (which would be a sibling to `meta.fixable`, and following the same naming convention as `fixable`) and make it mandatory when a rule provides suggestions. Eliminate `meta.docs.suggestion`. This is my ideal design.
2. Keep `meta.docs.suggestion` and make it mandatory when a rule provides suggestions.

As I mentioned in the PR description, the benefits of making the field mandatory include:
* Having static properties make it easy for both humans and tooling to tell whether rules provides autofixes or suggestions. For example, these properties can make it convenient to write tests that rules with autofixes/suggestions mention that they have autofixes/suggestions in their documentation.
* The question of whether a rule provides suggestions is a factual question with only one answer, so it's odd we would leave the `suggestion` property as optional when it would be straightforward for a rule to provide it (unlike the other properties in `meta.docs` which are more free-form fields).
* The `suggestion` property is essentially useless if it's presence is not enforced as neither humans nor tooling can actually rely on it.

Would you see a path forward to making an improvement here? I realize any improvement would be a breaking change, and I would be happy to write an RFC with a proposal.

Reply:
Anything under “docs” is intended to be optional and used only for generating the website. I don’t think it’s presence or absence should indicate anything. 

We have the “fixable” top-level property because we want to know if a rule can apply fixes because we can optimize when rules run. There’s no such concern for suggestions because they don’t make any changes to the source code.

So, I’m not sure we should do anything here except maybe update the documentation to reflect reality.

Reply:
Even though we don't currently have a concrete need for knowing whether rules provide suggestions, I guess I just personally wish that basic rule information like that was conveniently and reliably available to anyone interested to make use of it.

But if we won't consider changing this, then we can close this PR. I opened a separate PR #14298 to extract the documentation improvements :)

Reply:
I went ahead and opened a PR to create a lint rule called [eslint-plugin/require-meta-docs-suggestion](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/pull/105) for anyone interested to enforce the presence of the `meta.docs.suggestion` property for suggestable rules.

Reply:
We can consider changing it, but it needs more discussion. If you want to pursue it, I’d suggest opening an issue to start the discussion as such a change would need an RFC before being implemented.

Reply:
Sounds good, I opened this issue to discuss: #14312.